R/reuseMeta.R

```diff
@@ -21,15 +21,38 @@
 #'# see createGADS vignette
 #'
 #'@export
-reuseMeta <- function(GADSdat, varName, other_GADSdat, other_varName = NULL, missingLabels = NULL, addValueLabels = FALSE) {
+reuseMeta <- function(GADSdat, varName, other_GADSdat, other_varName = NULL,
+                      missingLabels = NULL, addValueLabels = FALSE) {
   UseMethod("reuseMeta")
 }
 #'@export
-reuseMeta.GADSdat <- function(GADSdat, varName, other_GADSdat, other_varName = NULL, missingLabels = NULL, addValueLabels = FALSE) {
-  if(!is.null(missingLabels) && !missingLabels %in% c("drop", "leave", "only")) stop("Invalid input for argument missingLabels.")
-  if(!varName %in% names(GADSdat$dat)) stop("varName is not a variable in the GADSdat.")
-  # extract meta data
+reuseMeta.GADSdat <- function(GADSdat, varName, other_GADSdat, other_varName = NULL,
+                              missingLabels = NULL, addValueLabels = FALSE) {
+  check_GADSdat(GADSdat)
+  check_vars_in_GADSdat(GADSdat, vars = varName)
+  if(inherits(other_GADSdat, "GADSdat")) check_GADSdat(other_GADSdat)
+  if(inherits(other_GADSdat, "all_GADSdat")) check_all_GADSdat(other_GADSdat)
   if(is.null(other_varName)) other_varName <- varName
+  check_vars_in_GADSdat(other_GADSdat, vars = other_varName)
+  if(length(varName) != length(other_varName)) {
+    stop("'varName' and 'other_varName' have different length.")

```

Comment:
Nit: "different length**s**"?

Reply:
Mh, it was a conscious choice, similar to this:
https://english.stackexchange.com/questions/7820/of-different-size-or-of-different-sizes

Do you think it sounds very 'wrong'?

Reply:
Sorry, I did not do the research - does not sound wrong, but I was not sure about this. But the explanation makes sense to me! 👌